### ISSUE TITLE
ToNDArray()  "Given shape size (198000) does not match the size of the given storage size (264000)'"

### ISSUE THREAD

**Issue by unominhtuan** [SciSharp/NumSharp#358]
This is the code I use

![1](https://user-images.githubusercontent.com/43975116/66219858-23e40480-e6f6-11e9-8b26-d6d53f0fd548.png)
![2](https://user-images.githubusercontent.com/43975116/66219870-29d9e580-e6f6-11e9-9234-2a6a45f72a43.png)

and this is  "faster_rcnn_inception_v2_coco.config"  file:
[https://drive.google.com/open?id=1qLU2KYi9eepkKJ4OOYhj0Do9yzTKPNTy](url)
My input image is sized (400,165)

I use ReadTensorFromFile (file_name), everything works fine, but it is very slow because I have to save pictures continuously to the hard drive.
![Image224114-04102019](https://user-images.githubusercontent.com/43975116/66220914-1e87b980-e6f8-11e9-8f5b-3f015620ed43.png)

Is there any solution that can convert NdArray from Bitmap? Or am I wrong somewhere during training?
Thanks very much.

**Comment by Nucs** [comment]
It might be with the image you are loading, can I have an image that you try to load which causes the error?

**Comment by unominhtuan** [comment]
> It might be with the image you are loading, can I have an image that you try to load which causes the error?

![3](https://user-images.githubusercontent.com/43975116/66221927-352f1000-e6fa-11e9-943c-bf373074febc.jpg)
![4](https://user-images.githubusercontent.com/43975116/66221928-352f1000-e6fa-11e9-8c7c-17cf0b179f12.jpg)
![1](https://user-images.githubusercontent.com/43975116/66221929-35c7a680-e6fa-11e9-945d-110f9231961e.jpg)
![2](https://user-images.githubusercontent.com/43975116/66221930-35c7a680-e6fa-11e9-9b50-6e646b3ffebd.jpg)

**Comment by Nucs** [comment]
Basically 198,000 is 3 pixels image and 264,000 is 4 pixels image. Meaning the image you are loading is likely to support transparency (alpha pixel).
I'll look into it tommorow first thing in the morning.

**Comment by unominhtuan** [comment]
@Nucs Thanks for taking the time

**Comment by unominhtuan** [comment]
@Nucs  Thanks a lot, 
I solved the problem by converting photos to PixelFormat.Format24bppRgb before using ToNDArray(). Thanks for suggesting me on transparency image.

**Comment by Nucs** [comment]
I'll close this once I present a solution on the library.

**Comment by Nucs** [comment]
I've released [0.20.4 for NumSharp.Bitmap](https://www.nuget.org/packages/NumSharp.Bitmap/0.20.4).
One of the related changes is you can do now: 
```C#
var nd = bitmap.ToNDArray(discardAlpha: true); //returns ndarray shaped (1, height, width, 3) instead of (..., 4).
```
